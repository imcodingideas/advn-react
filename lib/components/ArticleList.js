
import React from 'react';

import Article from './Article';

class ArticleList extends React.Component {
  render() {
    return (
      <div>
        {this.props.articles.map(article =>
          <Article key={article.id}
            onClick={() => { console.log('test'); }}
            article={article} />
        )}
      </div>
    );
  }
}

// articles: []

export default ArticleList;