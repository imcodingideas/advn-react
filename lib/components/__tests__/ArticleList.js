import React from 'react';
import ArticleList from '../ArticleList';

import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const articles = [
    { id: 1, author: {} },
    { id: 2, author: {} },
    { id: 3, author: {} },
  ];

  const tree = renderer.create(
    <ArticleList articles={articles} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
  expect(tree.children.length).toBe(3);
});