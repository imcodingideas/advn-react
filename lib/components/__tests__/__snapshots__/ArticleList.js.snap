// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <div />
    <div />
    <div>
      <div>
        undefined, undefined
      </div>
      <div />
    </div>
    <div />
  </div>
  <div>
    <div />
    <div />
    <div>
      <div>
        undefined, undefined
      </div>
      <div />
    </div>
    <div />
  </div>
  <div>
    <div />
    <div />
    <div>
      <div>
        undefined, undefined
      </div>
      <div />
    </div>
    <div />
  </div>
</div>
`;
